# Лабораторная 2 Численные методы

Вариант 5

Рассмотрим набор различных точек на отрезке $[a,b]:\ x_0 < x_1 < \dots < x_n$, в которых заданы значения функции $f(x)$ так, что $f_i=f(x_i), i=0,1,\dots ,n$.Требуется восстановить значение $f(x)$ и в других точках $x \in [a,b]$.


Для построения таблицы $(x_i, f_i)$ дано:

$[a,b] = [0.35; 1.35]$,  $x_i=0.35 + i \cdot h,\ h=\frac{1}{10},\ i=0,1,\dots,n$, где $n=10$.

$f(x)=0.35 e^x + 0.65 \cos x$


Необходимо:
1. Построить элемент наилучшего приближения методом наименьших квадратов.
2. Построить интерполяционный многочлен Лагранжа.
3. Построить остаток интерполирования в форме Ньютона.
4. Минимизировать остаток интерполирования через точки многочлена Чебышева.

### Вычислим известные значения функций в узлах $x_i$:

In [10]:
import math

def foo(x: float)->float:
    return 0.35 * math.exp(x) + 0.65 * math.cos(x)

nodes: list[float] = [foo(x / 100) for x in range(35, 136, 10)]
nodes

[1.1072659053584362,
 1.1341998814507992,
 1.160779495592267,
 1.1878937592117498,
 1.2165477705824197,
 1.2478654429993352,
 1.2830923889120704,
 1.32359907245173,
 1.3708843549661207,
 1.426579570668569,
 1.4924532823544179]

|i |0|1|2|3|4|5|
|--|--|--|--|--|--|--|
|x_i |0.35|0.45|0.55|0.65|0.75|0.85|
|f(x_i) |1.1072659053584362| 1.1341998814507992 |  1.160779495592267 |1.1878937592117498|1.2165477705824197|1.2478654429993352|




|i |6|7|8|9|10|
|--|--|--|--|--|--|
|x_i |0.95|0.105|0.115|0.125|0.135|
|f(x_i) |1.2830923889120704| 1.32359907245173 |  1.3708843549661207 |1.426579570668569| 1.4924532823544179|

## 1. Построение элемента наилучшего приближения методом наименьших квадратов.

Алгоритм:

Строим полином степени $n=5$: $$P(x)= \sum^n_{i=0} c_i x^i$$

Коэффициенты $c_i$ находим из системы уравнений: $$\sum^n_{i=0} c_i (\sum^N_{j=0} p(x_j) x_j^{i+k}) = \sum^N_{j=0} p(x_j) f(x_j) x_j^k,\ k=\overline{0, n},\ p(x) \equiv 1$$